src/main/java/seedu/address/model/person/Availability.java

```
@@ -6,19 +6,22 @@
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
 
 /**
  * Represents a Person's availability in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAvailability(String)}
  */
 public class Availability {
     public static final String MESSAGE_CONSTRAINTS = "Availabilities must be in the format of dd/MM/yyyy";
+    public static final String MESSAGE_CONSTRAINTS_DATE = "Availabilities must be a valid date";
 
     /**
      * The first character of the availability must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
     public static final String VALIDATION_REGEX = "dd/MM/yyyy";
+    public static final String VALIDATION_REGEX_DATE = "\\d{2}/\\d{2}/\\d{4}";
```

Perhaps the `VALIDATION_REGEX_DATE` can have stricter constraints? Found one that might work [here](https://stackoverflow.com/questions/15491894/regex-to-validate-date-formats-dd-mm-yyyy-dd-mm-yyyy-dd-mm-yyyy-dd-mmm-yyyy). So that dates like `31/04/2024` should raise an error.

Can refer [here](https://regex101.com/library) for community regex patterns written by others. The same site has also features to test your regex easily.

---

Thanks, I have updated to fix this issue.